include/drivers/adc.h

```diff
@@ -348,6 +396,30 @@ typedef int (*adc_api_read_async)(const struct device *dev,
 				  const struct adc_sequence *sequence,
 				  struct k_poll_signal *async);
 
+/**
+ * @brief Type definition of ADC API function for configuring adc threshold
+ *        interruption.
+ * See adc_api_th_irq_config() for argument descriptions.
+ */
+typedef int (*adc_api_th_irq_config)(const struct device *dev,
```

Comment:
Set the API internal name to match the external API name `adc_api_config_threshold_irq`.

Reply:
> This could use more details about the intended use.
Addressing this on next CL

> Also, do any other ADC controllers support a threshold mechanism?
I could not find any threshold detection at ADC driver level, normally threshold detection are handled by implementation such as temperature sensors.

Reply:
Done.